### DISCUSSION TITLE
mesa-frames 0.1.0-alpha

### DISCUSSION THREAD

**Discussion by adamamer20** [projectmesa/mesa-frames#46] [General]
As we near the end of Google Summer of Code, here's a prioritized list of tasks that need to be completed before the deadline:

1. **GridPolars Implementation**: 
   - Initially, I implemented grid capacity using a list of `pl.ranges` with the idea of leveraging lazy evaluation for better performance. However, this approach is only advantageous when capacity is consistent across all cells. Reimplementing grid behavior with Polars expressions would require significant effort without a clear performance gain. Therefore, for now, I'd stick with NumPy and update the mixin to `PolarsMixin` as needed.

2. **Example Models**: 
   - We need to provide at least two complete example models, such as the **Sugarscape** and **Schelling models**, to demonstrate how to build models using mesa-frames.

3. **Documentation**: 
   - This includes an introductory tutorial, performance tips, FAQ, and the deployment of API documentation.

4. **Automatic Build and Publishing**: 
   - Set up an automatic build and publishing pipeline for the package.

5. **GPU Integration (Time Permitting)**: 
   - This should be relatively straightforward, as it primarily involves switching from `pandas` to `cudf` and from `numpy` to `cupy`.

6. **General Review and Bug Discovery**: 
   - Conduct a thorough review and address any found bugs.

I'd focus on these priorities and leave any additional features or improvements for after the GSoC period.

**Comment by rht** [comment]
Can't wait for the Sugarscape one. [This](https://github.com/FLAMEGPU/FLAMEGPU2-submodel-benchmark/blob/master/src/main.cu) is the code for FLAME 2 GPU Sugarscape. It is going to be much simpler in mesa-frames. We could write a paper for this later.

**Reply by adamamer20** [reply]
Do you really think it could be published as a paper? That would be awesome!
Also, do you think the sugarscape example should implement instantaneous growback (like FLAME2) or constant growback (like mesa-examples)?

**Reply by rht** [reply]
Yes, SamPy was very specific on epidemiology. There was https://github.com/VU-IVM/honeybees, but they used NumPy/CuPy. So mesa-frames is still the first to use DF for ABM.
I think it's fine to use instantaneous growback so that it is an apple vs apple benchmark with FLAME 2.
